string data
uint64 size